src/app/pages/sharing/smb/smb-form/smb-form.component.spec.ts

```diff
@@ -0,0 +1,449 @@
+import { IxExplorerHarness } from 'app/modules/ix-forms/components/ix-explorer/ix-explorer.harness';
+import { IxInputHarness } from 'app/modules/ix-forms/components/ix-input/ix-input.harness';
+import { IxSelectHarness } from 'app/modules/ix-forms/components/ix-select/ix-select.harness';
+import { IxFormsModule } from 'app/modules/ix-forms/ix-forms.module';
+import { IxFormHarness } from 'app/modules/ix-forms/testing/ix-form.harness';
+import { AppLoaderService, DialogService, WebSocketService } from 'app/services';
+import { FilesystemService } from 'app/services/filesystem.service';
+import { IxSlideInService } from 'app/services/ix-slide-in.service';
+import { SmbFormComponent } from './smb-form.component';
+
+describe('SmbFormComponent', () => {
+  const existingShare: SmbShare = {
+    id: 1,
+    purpose: 'ENHANCED_TIMEMACHINE',
+    path: '/mnt/pool123/ds222',
+    path_suffix: '%U',
+    home: false,
+    name: 'ds222',
+    comment: '',
+    ro: false,
+    browsable: true,
+    recyclebin: true,
+    guestok: true,
+    hostsallow: ['host1'],
+    hostsdeny: ['host2'],
+    auxsmbconf: '',
+    aapl_name_mangling: false,
+    abe: true,
+    acl: true,
+    durablehandle: true,
+    streams: true,
+    timemachine: true,
+    vuid: 'a7bcb6cb-b2f3-4144-a5bb-e79dc7e282c4',
+    shadowcopy: true,
+    fsrvp: false,
+    enabled: true,
+    cluster_volname: '',
+    locked: false,
+  };
+
+  const formLabels: { [key: string]: string } = {
+    path: 'Path',
+    name: 'Name',
+    purpose: 'Purpose',
+    comment: 'Description',
+    enabled: 'Enabled',
+    acl: 'Enable ACL',
+    ro: 'Export Read Only',
+    browsable: 'Browsable to Network Clients',
+    guestok: 'Allow Guest Access',
+    abe: 'Access Based Share Enumeration',
+    hostsallow: 'Hosts Allow',
+    hostsdeny: 'Hosts Deny',
+    home: 'Use as Home Share',
+    timemachine: 'Time Machine',
+    afp: 'Legacy AFP Compatibility',
+    shadowcopy: 'Enable Shadow Copies',
+    recyclebin: 'Export Recycle Bin',
+    aapl_name_mangling: 'Use Apple-style Character Encoding',
+    streams: 'Enable Alternate Data Streams',
+    durablehandle: 'Enable SMB2/3 Durable Handles',
+    fsrvp: 'Enable FSRVP',
+    path_suffix: 'Path Suffix',
+    auxsmbconf: 'Auxiliary Parameters',
+  };
+
+  const presets: SmbPresets = {
+    NO_PRESET: {
+      verbose_name: 'No presets',
+      params: {
+        auxsmbconf: '',
+      },
+    },
+    DEFAULT_SHARE: {
+      verbose_name: 'Default share parameters',
+      params: {
+        path_suffix: '',
+        home: false,
+        ro: false,
+        browsable: true,
+        timemachine: false,
+        recyclebin: false,
+        abe: false,
+        hostsallow: [],
+        hostsdeny: [],
+        aapl_name_mangling: false,
+        acl: true,
+        durablehandle: true,
+        shadowcopy: true,
+        streams: true,
+        fsrvp: false,
+        auxsmbconf: '',
+      },
+    },
+    ENHANCED_TIMEMACHINE: {
```

Comment:
Since this is coming from backend as far as I understand, one should be enough to test our code.